include/maliput_sparse/parser/parser.h

```diff
@@ -0,0 +1,63 @@
+// BSD 3-Clause License
+//
+// Copyright (c) 2022, Woven Planet.
+// All rights reserved.
+//
+// Redistribution and use in source and binary forms, with or without
+// modification, are permitted provided that the following conditions are met:
+//
+// * Redistributions of source code must retain the above copyright notice, this
+//   list of conditions and the following disclaimer.
+//
+// * Redistributions in binary form must reproduce the above copyright notice,
+//   this list of conditions and the following disclaimer in the documentation
+//   and/or other materials provided with the distribution.
+//
+// * Neither the name of the copyright holder nor the names of its
+//   contributors may be used to endorse or promote products derived from
+//   this software without specific prior written permission.
+//
+// THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS "AS IS"
+// AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO, THE
+// IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE ARE
+// DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT HOLDER OR CONTRIBUTORS BE LIABLE
+// FOR ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY, OR CONSEQUENTIAL
+// DAMAGES (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR
+// SERVICES; LOSS OF USE, DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER
+// CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY,
+// OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE
+// OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+#pragma once
+
+#include <unordered_map>
+
+#include <maliput/common/maliput_copyable.h>
+
+#include "maliput_sparse/parser/segment.h"
+
+namespace maliput_sparse {
+namespace parser {
+
+/// Pure virtual parser API class to be inherited by a specific parser class.
+/// @code {.cpp}
+/// class MyParser : public Parser {
+///  public:
+///   MyParser(const std::string& my_file) : Parser() { // Parse underlying file and fill up segments.}
+///   const std::unordered_map<Segment::Id, Segment>& DoGetSegments() const override {
+///     return segments_;
+///   }
+/// @endcode
+class Parser {
+ public:
+  MALIPUT_NO_COPY_NO_MOVE_NO_ASSIGN(Parser)
+  Parser() = default;
+  virtual ~Parser() = default;
+
+  /// Gets the segments parsed from the input description.
+  const std::unordered_map<Segment::Id, Segment>& GetSegments() const {return DoGetSegments();}
```

Comment:
I think it is a bit premature to move things here. This is an example of that. We would likely need also the Junction information and we are not even configuring other RoadNetwork entities (e.g. Intersections, Phases, etc.). The Parser should evolve to to provide all of them most likely.

I would refrain adding this class for the moment.